### DISCUSSION TITLE
Select server (too cetralised on matrix.org is causing slowdowns)

### DISCUSSION THREAD

**Discussion by marco-trovato** [element-hq/element-meta#375] [General]
### Have you considered any alternatives?

**Problem:**
The main matrix.org server is currently too slow and overloaded.
This is another form of "centralisation".

**Proposed solution:**
When creating a new account, please let the user chose another server from a dropdown with at least 2 or 3 alternative servers from the list: https://joinmatrix.org/servers

Simply open the web browser redirecting to the registration page would be a perfect solution.

**Comment by notramo** [comment]
It won't happen. Element was and will always be designed to promote centralisation.
If you don't want to use overloaded servers, promote alternative clients, like [SchildiChat](https://schildi.chat/), [Syphon](https://syphon.org), or [FluffyChat](https://fluffychat.im/), which make it easier to select alternative servers. This would also increase client app diversity.

**Comment by marco-trovato** [comment]
Very open source.

**Comment by notramo** [comment]
It's open source, for example SchildiChat is an Element for with these problems fixed.

**Comment by ouchadam** [comment]
thanks for raising! this is a topic that we're aware of and encapsulates all the element clients, we're still considering/figuring out the best way to approach incorporating other homeservers

**Comment by daniellekirkwood** [comment]
This is a really great discussion, I'm moving it to this repo and to discussions so that we don't lose the feedback and can continue to think about how we might address this.

**Comment by HarHarLinks** [comment]
I have said this before, on the issue trackers of the different Element clients among other places:
Regardless of it's performance, if matrix.org and Element really stand behind the decentralised ideals they claim to do, then they *must* move away from promoting matrix.org so heavily.
The FTUE has just now been redesigned (at least on Android) and in a way that in my opinion completely ignores this, and on the contrary reinforces the selection of the matrix.org server.

When I open my app the first time (F-Droid so not totally up to date because of proprietary library issues, so I hope this is sufficiently up to date, else I'll reevaluate this) and click create account, it presents to me a screen where
1. the first(!) option is joining matrix.org, the "largest public server". This is made to sound like an advertisement, when actually you best choice in the ecosystem would be close to 100% self-hosting. In actuality this description should discourage one from joining. It should be the last and fall-back uption. "You don't know? Then use this".
2. The second option is EMS, which isn't the most "free" option that I think would be ideal, however it is close to the decentralisation ideology and it's also an Element product which in my eyes is ok and probably even a good choice. It is almost like the "paid version of the app".
3. The third option is just a very generic "other" and "advanced settings", almost actively discouraging users from learning about self hosting(!). There is not a explanation about "homeservers" and self hosting on this page, nor is one linked. Just "learn more" about EMS. Why?
4. What's this? A fourth option exists, but with a completely different layout, and it offers to "Sign in", but why does it offer that when I've just selected to create an account? That this should be the primary and *most obvious* way to log in is a whole different discussion.

When I open app.element.io, the matrix.org server is not just prefilled, but in a text field that is locked behind an edit button.
![image](https://user-images.githubusercontent.com/2803622/173442805-df60a0d8-7174-4566-8933-8263dbfaa324.png)
Instead, it should encourage people to either learn about this or at least consciously choose the matrix.org server.
I realise onboarding is hard (for users) and you want to make it as simple as possible for people to join from both a business standpoint and an ecosystem standpoint. However in my opinion, the current system is designed in direct contradiction to the decentralisation claim.

**Comment by notramo** [comment]
What about adding an input field for server address, and a text:
> Ask your friends what server do they recommend (probably the one they use).
If your friends don't use Matrix yet, then you have these options:
- Select from list (joinmatrix.org)
- Use matrix.org (hosted by the developers of Matrix)

(The bullet points would be buttons.)

**Comment by ShadowJonathan** [comment]
FWIW, joinmatrix.org is not affiliated with matrix.org and element.io, and takes some opinionated decisions when curating its list that might not align with what Element and Matrix.org want.

This isnt about there being a (inferred and speculated) conflict of interest when Element and/or the Matrix Foundation promotes decentralization, but that there's a liability concern when matrix.org promotes a third-party site as an "official" list, that might or might not act out in a way that the foundation deems non-constructive.

I'm not saying that there is a conflict like that *right now*, but that ive seen this happen at least a few times, where the community aligns with the wider interests of a business or a non-profit, and that they eventually disalign due to reasons, but the one still promotes the other, which causes strife when the community decides to boycot or hijack that promotion.

I think due to these reasons, joinmatrix.org would either have to become a joint/open project with foundation supervision, or another project has to emerge that does that, before the foundation would feel comfortable promoting such a thing, as things stand right now.

---

The situation with matrix.org itself is very tangled in an unfortunate way; `matrix.org` (the server) is ran by Element (or officially, New Vector Ltd.) as a donation/sponsor service.

The long-term intention is to get matrix to a point (with homeserver diversity, and/or with nomadic user (migration)) where the matrix.org server can be "turned off", this was a clear stated goal from the beginning, from high up.

However, due to the nature of Element positioning itself as an alternative to the likes of Whatsapp, Slack, Discord, Telegram, etc. they have to have competitive edges, and they barely have any to begin with.

Any friction that exists when a user signs up is a lost user, as they'll lose interest, and so the first few minutes after a user logs into ~~the~~ matrix is crucial to have them stay in, it's what Element has been focusing on across their clients in the last couple of months; retention.

`matrix.org` being "the default server" has sort-of been baked into the client from the beginning, as in the beginning it was a 'seed' server to get the federation started, but as there was no real way to migrate users across servers, matrix.org stayed full, and the Element team had bigger fish to fry at that time. Only recently had they really had the energy to look at this problem (and many others) with fresh eyes, thus this discussion point.

The real problem Element has here is balancing "useful" user friction by having them choose a server, with user signup retention, with user education; many of these users were never offered a choice of servers to sign up "with", many dont care, and most of them dont even know where to begin to look for "servers to sign up with". (which is comparable to gmail and hotmail, people just 'want email', they dont want difficulties)

---

Disclaimer: I do not work for the foundation, and am only extrapolating from what I've heard and what i think the foundation feels on these topics.

The foundation wants to trust, but due to events in the past, this has become rather difficult for them.

**Comment by notramo** [comment]
In this case, what about this text in the UI
> Ask your friends what server do they recommend (probably the one they are using).
If your friends aren't on Matrix yet, then try *matrix.org*
<hr/>
Easy registration is important, but later on, when they notice that matrix.org is slow, and features are disabled (e.g. presence), then it can be extra hassle to migrate to another server than it was if they registered originally on that server.

**Considering these points, the best solution for this would be probably to land account migration in the spec.**

**Comment by HarHarLinks** [comment]
> The situation with matrix.org itself is very tangled in an unfortunate way; `matrix.org` (the server) is ran by Element (or officially, New Vector Ltd.) as a donation/sponsor service.

This situation would be improved in my eyes if instead Element hosted (another) free `element.io` or `ems.element.io` service that is associated with the company rather than the organisation.

> The long-term intention is to get matrix to a point (with homeserver diversity, and/or with nomadic user (migration)) where the matrix.org server can be "turned off", this was a clear stated goal from the beginning, from high up.

Pretty much like I said in my earlier post. Let me rephrase to match your explanation more precisely: This goal is not going to be reached when Element as the most popular client that is tightly entangled with matrix.org keeps promoting a central server so heavily.

Hence my argument earlier that it would be kind of valid to even push EMS more strongly (no I don't want advertisement pop-ups), let alone the part where users are educated about (the ability) to self-host.
Multiple organisations have been invited to talk at OTWSU, an event organised by matrix.org, that offer ready made software to host your server, and I think enough organisational integrity etc for matrix.org to recommend them.

**Reply by ShadowJonathan** [reply]
This sounds like a good point, I havent heard of OTWSU offering this, do you have some further information on that?

**Reply by HarHarLinks** [reply]
Among the presenters have been https://yunohost.org and https://framasoft.org have been featured multiple times, including their chatons project https://www.chatons.org/en/search/by-service?service_type_target_id=118&field_alternatives_aux_services_target_id=All&field_software_target_id=All&field_is_shared_value=All&title=

OTWSU topics are archived at https://matrix.org/open-tech-will-save-us/

The [ansible playbook](https://github.com/spantaleev/matrix-docker-ansible-deploy) also makes it extremely easy to host a server with all kinds of bells and whistles, though its approach seems a bit more technical than yunohost, for example.

**Comment by ghost** [comment]
I will give you my opinion about this topic, In my opinion, Matrix and Element should create a curated list of good and reliable homeservers (around 50 more or less) and then, when the user creates an account, the client (Element) randomly picks one of them, then, prompts the user for approval. This system would probably work well, considering that the list is good and that the servers are truly reliable.

**Comment by paperwriter0300** [comment]
Exploring an education system without age restrictions is an exciting concept. It could open up opportunities for lifelong learning, allowing individuals to pursue their passions and interests  [https://researchpaperwriter.org/](https://researchpaperwriter.org/) at any stage in life, promoting personal growth and skill development.

Removing age restrictions from education could also foster intergenerational learning, where people of different age groups come together to exchange knowledge and experiences, enriching the learning environment for everyone involved.
